functions/packages.bash

```diff
@@ -1,103 +1,142 @@
+
+  echo -n "$(timestamp) [openHABian] Generating FireMotD theme... "
+  if cond_redirect FireMotD -G Gray; then echo "OK"; else echo "FAILED"; return 1; fi
+
+  if ! grep -q "FireMotD" /home/"${username:-openhabian}"/.bash_profile; then
+    echo -n "$(timestamp) [openHABian] Make FireMotD display on login... "
+    if echo -e "\\necho\\nFireMotD --theme Gray \\necho" >> /home/"${username:-openhabian}"/.bash_profile; then echo "OK"; else echo "FAILED"; return 1; fi
+  fi
+
+  echo "$(timestamp) [openHABian] Setting up FireMotD apt updates count service... "
+  cond_echo "Make FireMotD check for new updates every night... "
   echo "# FireMotD system updates check (randomly execute between 0:00:00 and 5:59:59)" > /etc/cron.d/firemotd
   echo "0 0 * * * root perl -e 'sleep int(rand(21600))' && /bin/bash /usr/local/bin/FireMotD -S &>/dev/null" >> /etc/cron.d/firemotd
-  # invoke apt updates check after every apt action ('apt-get upgrade', ...)
+  cond_echo "Make FireMotD check for new updates after using apt... "
   echo "DPkg::Post-Invoke { \"if [ -x /usr/local/bin/FireMotD ]; then echo -n 'Updating FireMotD available updates count ... '; /bin/bash /usr/local/bin/FireMotD --skiprepoupdate -S; echo ''; fi\"; };" > /etc/apt/apt.conf.d/15firemotd
-  echo "OK"
+  cond_echo "Initial FireMotD updates check"
+  if cond_redirect FireMotD -S; then echo "OK"; else echo "FAILED"; return 1; fi
+}
+
+exim_setup() {
+  echo -n "$(timestamp) [openHABian] Installing MTA required packages... "
+  if cond_redirect apt-get install --yes exim4 dnsutils mailutils; then
+    echo "OK"
+    create_mta_config
+  else
+    echo "FAILED"
+    return 1
+  fi
 }
 
 create_mta_config() {
-  TMP="$(mktemp /tmp/openhabian.XXXXX)"
-  HOSTNAME=$(/bin/hostname)
-  INTERFACES="$(/usr/bin/dig +short "$HOSTNAME" | /usr/bin/tr '\n' ';');127.0.0.1;::1"
-  EXIM_CLIENTCONFIG=/etc/exim4/passwd.client
-
-  introtext1="We will guide you through the install of exim4 as the mail transfer agent on your system and configure it to relay mails through a public service such as Google gmail.\\nPlease
-enter the data shown in the next window when being asked for. You will be able to repeat the whole installation if required by selecting the openHABian menu for MTA again."
-  introtext2="Mail server type: mail sent by smarthost; received via SMTP or fetchmail\\nSystem mail name: FQDN (your full hostname including the domain part)\\nIPs should be allowed by the se
-rver: $INTERFACES\\nOther destinations for which mail is accepted: $HOSTNAME\\nMachines to relay mail for: Leave empty\\nIP address or host name of outgoing smarthost: smtp.gmail.com::587\\nHide
- local mail name in outgoing mail: No\\nKeep number of DNS-queries minimal: No\\nDelivery method: Select: Maildir format in home directory\\nMinimize number of DNS queries: No\\nSplit configurat
-ion into small files: Yes\\n"
+  local eximConfig
```

Comment:
did you test mta install?
I think you didn't change anything substantial, did you? Moving large parts makes that hard to see.

Reply:
I have not yet, but for the most part all of the rewrite was using the same overall structure but safer and more verbose execution.

Reply:
That's alright. I just cannot validate it's still working like before by just looking at this diff so I'd expect you to execute real tests for it, too.